Set Glib2 minimum version to 2.32

gaborznagy opened the pull request:
Main motivation is to be able to use new features (embedded GMutex and [g_mutex_init()](https://developer.gnome.org/glib/stable/glib-Threads.html#g-mutex-init)) and remove long deprecated features (GStaticMutex).

Centos 6 will be EOL on 30 Nov 2020. I think platform support is not going to be affected. I've seen [AIX packages](https://repology.org/project/glib/versions) and  SLES 12 packages (from OBS builds) as well with much higher glib2 versions.

kira-syslogng commented:
Build SUCCESS

czanik commented:
Just checked, SLES SP4:
```
~> rpm -qa | grep -i glib2
glib2-devel-2.48.2-10.2.x86_64
glib2-tools-2.48.2-10.2.x86_64
```
Earlier service packs might feature earlier glib2 versions, but as far as I'm concerned, the majority of users are always running the latest service pack.

kira-syslogng commented:
Build SUCCESS

MrAnno commented:
Related commit that suppressed deprecation warnings:
https://github.com/syslog-ng/syslog-ng/commit/7d9feb8b355340b5b52f6d3e3dc55b51fa1e777b

gaborznagy commented:
So far, I've only found `GStaticMutex` which is deprecated.
```
 warning: ‘GStaticMutex’ is deprecated: Use 'GMutex' instead [-Wdeprecated-declarations]
```

We have a `GLIB-DISABLE-DEPRECATION-WARNINGS` macro to turn off deprecated glib APIs though...
https://developer.gnome.org/glib/stable/glib-Version-Information.html#GLIB-DISABLE-DEPRECATION-WARNINGS:CAPS
https://github.com/syslog-ng/syslog-ng/blob/66d8017a1198b15df72a58133756efee3dfb2755/lib/compat/glib.h#L29

gaborznagy commented:
Removing `GLIB_DISABLE_DEPRECATION_WARNINGS` reveals other deprecated APIs: e.g. `g_thread_init` or `g_cond_new`.

kira-syslogng commented:
Build FAILURE

kira-syslogng commented:
Build FAILURE

kira-syslogng commented:
Build SUCCESS

kira-syslogng commented:
Build SUCCESS

gaborznagy commented:
At last I've removed the Draft flag from this PR! :fireworks: :tada: :tada:

I'm doing some internal tests, please don't merge it until then! I keep you posted on the results!

Also, for the reviewers: should we split this PR into 2 PRs? It's quite huge.

kira-syslogng commented:
Build SUCCESS

kira-syslogng commented:
Build SUCCESS

gaborznagy commented on lib/compat/glib.h:
```diff
@@ -26,8 +26,6 @@
 
-#define GLIB_DISABLE_DEPRECATION_WARNINGS 1
-
 #include <glib.h>
```
Thanks for finding these! I would add it in this PR, as we won't deal with glib deprecations regularly.

szemere replied:
I just started to collect these functions when I found your comment. I can agree to put these findings (and hopefully others) into a separate PR in order to merge this one sooner.

gaborznagy replied:
Thanks again. I've fixed those and removed also some conditional calls to deprecated glib functions, and unnecessary version-guards. See 5bc10bb.

gaborznagy commented:
I've added a new patch about removing `GLIB_VERSION_MAX_ALLOWED` https://github.com/syslog-ng/syslog-ng/pull/3488/commits/5da6c8dba1789b4eb0f66687269c13ba09028634.
https://developer-old.gnome.org/glib/stable/glib-Version-Information.html#GLIB-VERSION-MAX-ALLOWED

gaborznagy commented:
I'v forgot to bump the min version in CMakeLists.txt, I've only bumped the GLIB_VERSION_MIN_REQUIRED macro.
Thanks @szemere for finding this!

P.s.: @szemere discovered that FindGLIB.cmake doesn't really work: cmake passes with any `version` value in `find_package(GLIB ...)` .

kira-syslogng commented:
Build SUCCESS

MrAnno commented:
> 
> 
> I'v forgot to bump the min version in CMakeLists.txt, I've only bumped the GLIB_VERSION_MIN_REQUIRED macro. Thanks @szemere for finding this!
> 
> P.s.: @szemere discovered that FindGLIB.cmake doesn't really work: cmake passes with any `version` value in `find_package(GLIB ...)` .

Changing the order of the parameters passed to `find_package()` seemed to fix the problem:
```
find_package(GLIB 2.32.0 REQUIRED COMPONENTS gmodule gthread)
```

szemere reviewed (dismissed):
LGTM!

Optional from my side:

- [x] @Kokan 's comment about other compat functions.
- [ ] reorder the version number in find_package

gaborznagy commented:
> > I'v forgot to bump the min version in CMakeLists.txt, I've only bumped the GLIB_VERSION_MIN_REQUIRED macro. Thanks @szemere for finding this!
> > P.s.: @szemere discovered that FindGLIB.cmake doesn't really work: cmake passes with any `version` value in `find_package(GLIB ...)` .
> 
> Changing the order of the arguments passed to `find_package()` seemed to fix the problem:
> 
> ```
> find_package(GLIB 2.32.0 REQUIRED COMPONENTS gmodule gthread)
> ```
> 
> https://cmake.org/cmake/help/v3.22/command/find_package.html#id5

Thanks! I think there should be no order in optional arguments, and this documentation doesn't emphasize that either. I'll just switch the order of arguments then (also switching the order of `REQUIRED` option too).
```
find_package(<PackageName> [version] [EXACT] [QUIET] [MODULE]
             [REQUIRED] [[COMPONENTS] [components...]]
             [OPTIONAL_COMPONENTS components...]
             [NO_POLICY_SCOPE])
```

MrAnno commented:
Yep, you are right:
The only positional arguments can appear at the beginning of the argument list. Anything without a dedicated KEYWORD-like argument signifier can be assumed to be positional.

We should fix this later. `pkg_check_modules()` now supports modern imported targets, we could use that instead of these huge and fragile find-modules of ours.

kira-syslogng commented:
Build SUCCESS

gaborznagy commented:
I've addressed the review comments, and finalized the last WIP commit (see https://github.com/syslog-ng/syslog-ng/commit/5bc10bb811b69ad49cdcad1ffb94dedbd591e881)

For the additonally removed deprecated functions, see 3f16b05

@Kokan @szemere please re-review this PR! Hopefully, only checking the diff of the last force-push helps.

Thank you for the thorough review! :+1:

gaborznagy commented:
I've forgot: I've done some internal tests, and everything seems to be alright. This PR is ready for merge.

szemere reviewed (approved):
Thank You for the changes!

bazsi commented:
\o/


On Fri, Nov 5, 2021 at 5:17 PM László Szemere ***@***.***>
wrote:

> ***@***.**** approved this pull request.
>
> Thank You for the changes!
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/syslog-ng/syslog-ng/pull/3488#pullrequestreview-799086436>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAFOK5VNF3XLAXTUYZZEY7DUKP7TBANCNFSM4TYXDZ6Q>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>


-- 
Bazsi
